src/aws_extension.cpp

```diff
@@ -67,19 +108,36 @@ static void LoadAWSCredentialsFun(ClientContext &context, TableFunctionInput &da
 		throw MissingExtensionException("httpfs extension is required for load_aws_credentials");
 	}
 
-	//! Return the Key ID of the key we found, or NULL if none was found
-	auto key_loaded = TrySetAwsCredentials(DBConfig::GetConfig(context), data.profile_name);
-	auto ret_val = !key_loaded.empty() ? Value(key_loaded) : Value(nullptr);
-	output.SetValue(0,0,ret_val);
+	auto load_result = TrySetAwsCredentials(DBConfig::GetConfig(context), data.profile_name, data.set_region);
+
+	// Set return values for all modified params
+	output.SetValue(0,0, load_result.set_access_key_id.empty() ? Value(nullptr) : load_result.set_access_key_id);
+	if (data.redact_secret && !load_result.set_secret_access_key.empty()) {
```

Comment:
Why do we not redact the secret when it's empty?
saying that the secret is null is kind of giving away information I feel

Reply:
Well when the credentials are invalid or empty as reported by `credentials.IsExpiredOrEmpty()` we don't set them. We return null to indicate the setting wasn't changed